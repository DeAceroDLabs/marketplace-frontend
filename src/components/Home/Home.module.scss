@use 'styles/colors';
@use 'styles/typography';

.header-title {
  color: colors.$neutral-inverse;
  font-size: typography.$header-m-size;
  font-weight: typography.$header-m-weight;
  z-index: 1;
}

.header-content {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
}

.header-content::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 100%;
  background: linear-gradient(180deg, rgba(0, 0, 0, 0.65) 19.38%, rgba(0, 0, 0, 0.3) 52.15%, rgba(0, 0, 0, 0.05) 88.61%);
}

.see-more-container {
  display: flex;
  justify-content: flex-end;
  margin: 10px 5%;

  .see-more {
    color: colors.$accent;
  }
}