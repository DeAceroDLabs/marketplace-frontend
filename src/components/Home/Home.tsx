import React, { useEffect, useState } from "react";
import { fetchCategories, fetchProducts } from "config/api";
import { Product } from "config/api.types";
import BgSection from "components/common/BgSection";
import Card from "components/common/Card";
import GridContainer from "components/common/GridContainer";
import Section from "components/common/Section";
import Tabs from "components/common/Tabs";
import View from "components/common/View";
import styles from "./Home.module.scss";
import { TabItem } from "components/common/common.types";
import SearchBar from "components/common/SearchBar";

const Home: React.FunctionComponent = () => {
  const [categories, setCategories] = useState<TabItem[]>([]);
  const [products, setProducts] = useState<Product[]>([]);
  const [activeCategory, setActiveCategory] = useState({} as TabItem);

  useEffect(() => {
    fetchCategories().then((data) => {
      const categories = data.map(
        (category) =>
          ({
            id: category.IdCategoria,
            title: category.NomCategoria,
          } as TabItem)
      );
      setCategories(categories);
    });
  }, []);

  useEffect(() => {
    activeCategory?.id &&
      fetchProducts(activeCategory.id).then((data) => {
        const products = data;
        setProducts(products);
      });
  }, [activeCategory, categories]);

  const setCategoryActive = (category: TabItem) => {
    setActiveCategory(category);
  };

  const productsCards = products.map((product) => {
    return (
      <Card
        key={product.ClaProducto}
        title={product.NomProducto}
        imgSrc={product.CategoriaUrlImagen}
      />
    );
  });

  const homeHeader = (
    <BgSection bgImage="/background.png">
      <div className={styles["header-content"]}>
        <div className={styles["header-title"]}>
          Somos tu aliado de confianza para hacer crecer tu negocio
        </div>
        <SearchBar />
      </div>
      <div className={styles.red}></div>
    </BgSection>
  );

  return (
    <View header={homeHeader}>
      <Section title="CatÃ¡logo de Productos">
        <Tabs
          options={categories}
          variant="secondary"
          onSelectTab={setCategoryActive}
        />
        <GridContainer>{productsCards}</GridContainer>
      </Section>
    </View>
  );
};

export default Home;
