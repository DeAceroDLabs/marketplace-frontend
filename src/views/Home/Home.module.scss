@use 'styles/colors';
@use 'styles/typography';
.header-title {
    color: colors.$color-background-01;
    font-size: typography.$header-m;
    font-weight: typography.$bold;
    z-index: 1;
}

.header-content {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100%;
}

.header-content::after {
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    background: linear-gradient(180deg, rgba(0, 0, 0, 0.65) 19.38%, rgba(0, 0, 0, 0.3) 52.15%, rgba(0, 0, 0, 0.05) 88.61%);
}

.products-container {
    width: fit-content;
    margin: auto;
}

.see-more-container {
    display: flex;
    justify-content: flex-end;
    margin: 10px 5%;
    .see-more {
        color: colors.$accent;
    }
}

.arrow-categories-container{
    display: flex;
    flex-direction: row;
    .button-container {
        margin-bottom: 40px;
        Button {
            position: absolute;
            font-size: typography.$body-m;
        }
    }
    
    .arrow-button {
        Button {
            padding: 13px;
            margin-right: 5px;
        }
        .icon {
            font-size: typography.$body-m;
        }

    }
    .next{
        margin-left: 20px;
    }
    .back{
        margin-right: 40px;
    }
}