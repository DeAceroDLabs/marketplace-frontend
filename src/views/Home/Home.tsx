import React, { useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";
import { fetchCategories, fetchProducts } from "config/api";
import { Product } from "config/api.types";
import BgSection from "components/common/BgSection";
import Card from "components/common/Card";
import LoadingCard from "components/common/LoadingCard";
import GridContainer from "components/common/GridContainer";
import Section from "components/common/Section";
import Tabs from "components/common/Tabs";
import View from "components/common/View";
import styles from "./Home.module.scss";
import { TabItem } from "components/common/common.types";
import SearchBar from "components/common/SearchBar";
import Button from "components/common/Button";
import ArrowBackIosNewIcon from "@mui/icons-material/ArrowBackIosNew";
import ArrowForwardIosIcon from "@mui/icons-material/ArrowForwardIos";

const Home: React.FunctionComponent = () => {
  const [categories, setCategories] = useState<TabItem[]>([]);
  const [products, setProducts] = useState<Product[]>([]);
  const [activeCategory, setActiveCategory] = useState({} as TabItem);
  const [loadingCards, setLoadingCards] = useState(false);
  let productsCards;
  const navigate = useNavigate();

  useEffect(() => {
    fetchCategories().then((data) => {
      const categories = data.map(
        (category) =>
          ({
            id: category.categoryId,
            title: category.categoryName,
          } as TabItem)
      );
      setCategories(categories);
    });
  }, []);

  useEffect(() => {
    setLoadingCards(true);
    activeCategory?.id &&
      fetchProducts(activeCategory.id).then((data) => {
        const products = data;
        setProducts(products);
        setLoadingCards(false);
      });
  }, [activeCategory, categories]);

  const setCategoryActive = (category: TabItem) => {
    setActiveCategory(category);
  };

  if (loadingCards) {
    productsCards = <LoadingCard cards={6} variant="small" />;
  } else {
    productsCards = products.slice(0, 6).map((product) => {
      return (
        <Card
          key={product.productId}
          title={product.productName}
          imgSrc={product.categoryImageUrl}
          size="small"
          loading={false}
        />
      );
    });
  }

  const homeHeader = (
    <BgSection bgImage="/background.png">
      <div className={styles["header-content"]}>
        <div className={styles["header-title"]}>
          Somos tu aliado de confianza para hacer crecer tu negocio
        </div>
        <SearchBar />
      </div>
    </BgSection>
  );

  const backArrow = categories.length ? (
    <div
      className={`${styles["button-container"]} ${styles["arrow-button"]} ${styles.back}`}
    >
      <Button color="primary" action={() => void 0} active={false}>
        <ArrowBackIosNewIcon fontSize="small" className={styles["icon"]} />
      </Button>
    </div>
  ) : null;

  const fowardArrow = categories.length ? (
    <div
      className={`${styles["button-container"]} ${styles["arrow-button"]} ${styles.next}`}
    >
      <Button color="primary" action={() => void 0} active={false}>
        <ArrowForwardIosIcon fontSize="small" className={styles["icon"]} />
      </Button>
    </div>
  ) : null;

  return (
    <View header={homeHeader}>
      <Section title="Catálogo de Productos">
        <div className={styles["arrow-categories-container"]}>
          {backArrow}
          <Tabs
            options={categories}
            variant="secondary"
            onSelectTab={setCategoryActive}
          />
          {fowardArrow}
        </div>
        <div className={styles["products-container"]}>
          <GridContainer>{productsCards}</GridContainer>
        </div>
        {products.length > 6 && (
          <div className={styles["see-more-container"]}>
            <Button
              color="transparent"
              action={() => {
                navigate(`/search/${activeCategory.title}`);
              }}
            >
              <div className={styles["see-more"]}>Ver más</div>
            </Button>
          </div>
        )}
      </Section>
    </View>
  );
};

export default Home;
