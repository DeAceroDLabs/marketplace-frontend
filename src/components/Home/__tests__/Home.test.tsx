import renderer from "react-test-renderer";
import Home from "../Home";

it("renders Home with no issue", () => {
  const component = renderer.create(<Home />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
