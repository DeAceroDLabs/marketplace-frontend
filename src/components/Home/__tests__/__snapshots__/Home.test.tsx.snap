// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home with no issue 1`] = `
Array [
  <div
    className="header"
  >
    <div>
      <div
        className="container horizontal"
      >
        <img
          alt=""
          className="bgImage"
          src="/background.png"
        />
        <div
          className="content"
        >
          <div
            className="header-content"
          >
            <div
              className="header-title"
            >
              Somos tu aliado de confianza para hacer crecer tu negocio
            </div>
          </div>
          <div
            className="red"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="body"
  >
    <div>
      <div
        className="section"
      >
        <div
          className="title"
        >
           
          Cat√°logo de Productos
           
        </div>
        <div
          className="body"
        >
           
          <div
            className="tabs"
          />
          <div
            className="container"
          />
           
        </div>
      </div>
    </div>
  </div>,
  <div
    className="footer"
  >
    <div />
  </div>,
]
`;
